jenkins/release-process/github-release.groovy

```diff
@@ -0,0 +1,106 @@
+                }
+            }
+        }
+        
+        stage("Create GitHub Release") {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
+                    withCredentials([string(credentialsId: 'gaurav-github-token', variable: 'GH_TOKEN')]) {
+                        env.github_release_url = sh( script: """
+                            bash scripts/release_support/create-cortx-github-release.sh -t $GIT_TAG -v $SERVICES_VERSION -c $CHANGESET_URL -r $RELEASE_REPO
+                        """, returnStdout: true).trim()
+                    }			
+                }
+            }
+	    }
+    }
+    post {
+        always {
+
+            script {
+
+                // Jenkins Summary
+                if ( currentBuild.currentResult == "SUCCESS" ) { 
+                    MESSAGE = "CORTX Release: ${IMAGE_TAGS} (Available on GitHub)"
+                    ICON = "accept.gif"
+                    STATUS = "SUCCESS"
+                } else if ( currentBuild.currentResult == "FAILURE" ) {
+                    manager.buildFailure()
+                    MESSAGE = "Failure: GitHub Release ${GIT_TAG}"
+                    ICON = "error.gif"
+                    STATUS = "FAILURE"
+ 
+                } else {
+                    manager.buildUnstable()
+                    MESSAGE = "GitHub release creation is Unstable"
+                    ICON = "warning.gif"
+                    STATUS = "UNSTABLE"
+                }
+
+                manager.createSummary("${ICON}").appendText("<h3>CORTX GitHub Release creation ${currentBuild.currentResult} </h3><p>Please check <a href=\"${BUILD_URL}/console\">GitHub release logs</a> for more info", false, false, false, "red")
+
+                // Email Notification
+                env.build_stage = "${build_stage}"
+                env.tags = "${IMAGE_TAGS}"
+
+                def toEmail = "gaurav.chaudhari@seagate.com"
+                def recipientProvidersClass = [[$class: 'DevelopersRecipientProvider']]
+                if ( manager.build.result.toString() == "FAILURE" ) {
+                    toEmail = "${toEmail}"
+                    recipientProvidersClass = [[$class: 'DevelopersRecipientProvider'], [$class: 'RequesterRecipientProvider']]
+                }
+               
+                emailext ( 
+                    body: '''${SCRIPT, template="github-release-email.template"}''',
+                    mimeType: 'text/html',
+                    subject: "${MESSAGE}",
+                    attachLog: true,
+                    to: toEmail,
+                    recipientProviders: recipientProvidersClass
+                )
+
+                archiveArtifacts artifacts: "api_response.html", onlyIfSuccessful: false, allowEmptyArchive: true
+            }
+            cleanWs()
+        }
+    }            
+}
+
+def getImageTags(image, tag) {
```

Comment:
Can we move this function to shell script?

Reply:
Moved and tested - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Release-Process/job/Create-GitHub-Release/94/